nwbinspector/nwbinspector.py

```diff
@@ -172,22 +197,21 @@ def inspect_nwb(
                     unorganized_results["PYNWB_VALIDATION"].append(message)
             nwbfile = io.read()
             check_results = list()
-            for importance, checks_per_object_type in checks.items():
+            for importance, check_functions in checks.items():
                 if importance.value >= importance_threshold.value:
-                    for check_object_type, check_functions in checks_per_object_type.items():
-                        for nwbfile_object in nwbfile.objects.values():
-                            if issubclass(type(nwbfile_object), check_object_type):
-                                for check_function in check_functions:
```

Comment:
I remember now; the original hope was that we would be able to do away with a level of iteration by directly subsetting the registry based on the NWBFile object's type; that is, `for check_function in check_functions[type(nwbfile_object)]`. That would have saved a certain amount of time from unnecessary `if` checks, however, when we started defining checks over general object types (`NWBContainer`, `TimeSeries`, etc.) this became impossible which is why we test `if issubclass`.

So yeah, I suppose we don't actually gain anything anymore by sub-indexing `Importance` per `neurodata_type`, so flattening the registry should be fine then.